activation/validation.go

```diff
@@ -62,31 +59,56 @@ func (v *Validator) NIPost(nodeId types.NodeID, commitmentAtxId types.ATXID, nip
 		return 0, err
 	}
 
-	var ref types.PoetProofRef
-	copy(ref[:], nipost.PostMetadata.Challenge)
-	proof, err := v.poetDb.GetProof(ref)
-	if err != nil {
-		return 0, fmt.Errorf("poet proof is not available %x: %w", nipost.PostMetadata.Challenge, err)
+	poetChallenge := nipost.Membership.Leaf
+	if !bytes.Equal(poetChallenge[:], expectedChallenge[:]) {
+		return 0, fmt.Errorf("invalid `Challenge` in membership merkle proof leaf; expected: %x, given: %x", expectedChallenge, poetChallenge)
```

Comment:
i think this check and additional 32 bytes are redundant, you can use nipost hash for merkle computation

Reply:
True. I will remove `Leaf` from `types.MerkleProof`